pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -512,15 +513,11 @@ func (ku *KubeUtil) init() error {
 	kubeletHttpsPort := config.Datadog.GetInt("kubernetes_https_kubelet_port")
 	kubeletHttpPort := config.Datadog.GetInt("kubernetes_http_kubelet_port")
 
-	hosts := host{ips: nil, hostnames: nil}
-	err = getPotentialKubeletHosts(kubeletHost, &hosts)
-	if err != nil {
-		return err
-	}
+	potentialHosts := getPotentialKubeletHosts(kubeletHost)
 
-	potentialKubeletHostsFilter(&hosts)
+	potentialKubeletHostsFilter(potentialHosts)
```

Comment:
nit: a better name for this method could be `FilterConnectionInfo` or `DedupeConnectionInfo`. What's important is not that it works on the `potentialHosts` variable, but that it deduplicates things in the `connectionInfo` type.